src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -84,25 +83,51 @@ public BattleshipModel() {
 
         scanResult = false;
     }
-
+// Refactored getShip into a case statement instead of a mass of cluttered if statements. -GH
     public Ship getShip(String shipName) {
-        if (shipName.equalsIgnoreCase("aircraftCarrier")) {
-            return aircraftCarrier;
-        } if(shipName.equalsIgnoreCase("battleship")) {
-            return battleship;
-        } if(shipName.equalsIgnoreCase("submarine")) {
-            return submarine;
-        } if(shipName.equalsIgnoreCase("clipper")) {
-            return clipper;
-        }if(shipName.equalsIgnoreCase("dinghy")) {
-            return dinghy;
-        } else {
-            return null;
+        switch(shipName) {
+            case "aircraftCarrier":
+                return aircraftCarrier;
+            case "battleship":
+                return battleship;
+            case "submarine":
+                return submarine;
+            case "clipper":
+                return clipper;
+            case "dinghy":
+                return dinghy;
+            default:
+                return null;
+        }
+    }
+public int enemyShipOrientation(boolean isEasy, int orientation){// This function returns the proper enemy ship orientation depending on game difficulty.
+        if(isEasy == true)
+            return 1;
+
+        else{
+            orientation = (Math.random() <= 0.5) ? 1 : 2;
```

Comment:
This notation can be a little difficult to understand at first glance, you may want to add a short comment in the future to explain essentially what this line is doing (something like: //generate a random number between 1 and 2).